data-reconciliation-app/src/services/authentication-service.ts

```diff
@@ -127,34 +208,69 @@ export class CertBasedAuthenticationService implements IAuthenticationService {
     }
   }
 
-  // 
-  public isValidIdentity(identityId: string): ServiceResult<boolean> {
-    if (!identityId) {
+  /**
+   * Check if caller's access token is valid
+   * @param {JwtAuthnIdentity} identity JwtAuthnIdentity object
+   * @returns {ServiceResult<boolean>}
+   */
+  public isValidJwtToken(identity: ccfapp.JwtAuthnIdentity): ServiceResult<boolean> {
+
+    if (!identity || !identity.jwt || !identity.jwt.payload || !identity.jwt.payload.sub) {
       return ServiceResult.Failed({
         errorMessage: "Error: invalid caller identity",
         errorType: "AuthenticationError",
       });
     }
 
-    const isMember = this.isActiveMember(identityId);
-    if (isMember.success && isMember.content) {
-      return ServiceResult.Succeeded(true);
+    if (identity.jwt.keyIssuer === "https://demo") { // logic to validate the tokens of demo issuer
+      // no further validation
     }
+    else if (identity.jwt.keyIssuer === "https://login.microsoftonline.com/common/v2.0") { // logic to validate the tokens microsoft idp issuer
+      // Microsoft identity platform access tokens
+      const msClaims = identity.jwt.payload as MSAccessToken;
+      
+      // check if token has the right version
+      if (msClaims.ver !== "1.0") {
+        return ServiceResult.Failed({
+          errorMessage: "Error: unsupported access token version, must be 1.0",
+          errorType: "AuthenticationError",
+        });
+      }
+
+      // Replace the below string with your own app id by registering an app in Azure:
+      // https://docs.microsoft.com/en-us/azure/active-directory/develop/quickstart-register-app
+      const MS_APP_ID = "ee48548a-7d69-4b8e-b2d4-805e8bac7f01";
+      const MS_APP_ID_URI = "api://b8dbd573-a015-424b-b111-2d5fa11cee3c";
```

Comment:
should we move this to `constants.ts` file?

Reply:
I kept it here for clarity.

Reply:
Done